src/components/Dashboard/Header/Authentication/Login.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import {
+  Form, Button, ButtonGroup, ToggleButton, Container, Row, Col,
+} from 'react-bootstrap';
+import { Redirect } from 'react-router-dom';
+import { Toaster, Intent } from '@blueprintjs/core';
+import Header from '../Header';
+import {db, facebookProvider} from '../../../../fire';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.authWithFacebook = this.authWithFacebook.bind(this);
+    this.authWithGoogle = this.authWithGoogle.bind(this);
+    this.authWithEmailPassword = this.authWithEmailPassword.bind(this);
+    this.state = {
+      redirect: false,
+    };
+  }
+
+  authWithFacebook() {
```

Comment:
Rather than adding multiple methods for what basically is only one feature (sign in), why not create only one method and dynamically dispatch authentication options based on the button id they called on?

Reply:
I'm going to do this on the next Sprint in scope of refactoring.

Reply:
I'm going to do this on the next Sprint in the scope of refactoring.